### PR TITLE
fix: remove `OpCode` enum and update `VMOperation`'s `op` field type

### PR DESCRIPTION
**Pull request by mouseless-eth**
## Motivation

Addresses #1902

Ensures that tracing with VmTrace enabled will continue to be future proof in the presence of any opcodes added/renamed when ethereum undergoes a new hardfork.

## Solution

~~Removed the ethers::core::types::OpCode enum by deleting [it's source file](https://github.com/gakonst/ethers-rs/blob/master/ethers-core/src/types/trace/opcodes.rs#L70)~~

~~Changed VMOperation's op field to be of type string.~~

Create new enum `ExecutedInstruction` that deserializes opcodes from VmTrace into either `ExectutedInstruction::Known(Opcode)` or `ExecutedInstruction::Unknown(String)`

## PR Checklist

-   [x] Added Tests
-   [x] Added Documentation
-   [ ] Updated the changelog

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
imo this is probably easier to maintain than an enum, so I'm supportive of this,

**Comment by gakonst** [comment]
Should we instead do `enum Opcode { Known(Opcode), Unknown(String)`?

**Comment by mouseless-eth** [comment]
> Should we instead do `enum Opcode { Known(Opcode), Unknown(String)`? I like the ability to match.

This route looks like a valid solution to unwanted serialization errors - I will work on implementing it, and we can test how it works.

File: ethers-core/src/types/trace/mod.rs [hunk 1]
```diff
@@ -208,4 +225,19 @@ mod tests {
...
     }
+
+    #[test]
+    fn test_deserialize_unknown_opcode() {
+        let example_opcodes = r#"["GAS", "CREATE2", "CUSTOMOP"]"#;
+        let parsed_opcodes: Vec<ExectuedInstruction> =
+            serde_json::from_str(example_opcodes).unwrap();
+        assert_eq!(
+            vec![
+                ExectuedInstruction::Known(Opcode::GAS),
```
Comment (gakonst) [inline ethers-core/src/types/trace/mod.rs:236]:
```suggestion
                ExecutedInstruction::Known(Opcode::GAS),
```

**Review by gakonst — CHANGES_REQUESTED** [review]
typo on the type, lgtm otherwise

**Review by mouseless-eth — COMMENTED** [review]
Good catch, typo has been fixed

**Review by mouseless-eth — COMMENTED** [review]
forgot to add files to prev comment

**Pull request merged**